cmd/main.go

```diff
@@ -63,16 +62,16 @@ func main() {
 		os.Exit(1)
 	}
 
-	// create runtime client
-	cl, err := configClient(cfg)
+	ctx := controllerruntime.SetupSignalHandler()
```

Comment:
I used the provided function from the controller-runtime library to handle the setup signal as it was almost the same as the logic that was originally here. This however also returns the cancel context.
I could have mimic the logic and make it more closer to the original version, but there is no point/benefit in doing that.